server/routes/curfewAddress/addResidentDetails.ts

```diff
@@ -34,45 +48,102 @@ export default class AddResidentDetailsRoutes {
         town: convertToTitleCase(requestSummary.address.town),
         postcode: requestSummary.address.postcode,
       },
+      mainResident,
+      otherResidents,
     })
   }
 
   POST = async (req: Request, res: Response): Promise<void> => {
-    const { forename, surname, phoneNumber, relation, dateOfBirth, age } = req.body
+    const { checkRequestId, prisonNumber } = req.params
+    const { residentId, forename, surname, phoneNumber, relation, prisonerName } = req.body
+
+    let { otherResident } = req.body
+
     validateRequest(req, () => {
       const validationErrors: FieldValidationError[] = []
 
       if (!forename) {
-        validationErrors.push({ field: 'forename', message: 'Enter a first name' })
+        validationErrors.push({ field: 'forename', message: 'Enter the main occupier’s first name' })
       }
 
       if (!surname) {
-        validationErrors.push({ field: 'surname', message: 'Enter a last name' })
+        validationErrors.push({ field: 'surname', message: 'Enter the main occupier’s last name' })
       }
 
       if (!relation) {
-        validationErrors.push({ field: 'relation', message: "Enter the resident's relation to the offender" })
+        validationErrors.push({
+          field: 'relation',
+          message: `Enter the main occupier’s relationship to ${prisonerName}`,
+        })
+      }
+
+      if (otherResident) {
+        otherResident = otherResident.filter((resident: OtherResident) => {
+          if (this.areAllFieldsEmpty(resident)) {
+            return false
+          }
+          return true
+        })
+
+        otherResident.forEach((resident: OtherResident, index: number) => {
+          if (this.areAllFieldsEmpty(resident)) {
```

Comment:
won't this have already filtered these out above?

Reply:
I'd suggest just filter them out as an initial step before the validation method.
```
const presentOtherResidents = otherResident.filter(resident => !this.areAllFieldsEmpty(resident))

validateRequest(....

if (presentOtherResidents.length)
```


This would be nicer than relying on mutation inside the validation method which is a bit confusing. 
It would mean you could remove this  `if (otherResident)` check too.